### DISCUSSION TITLE
02-terraform-intermediario/05-modules/s3_module/main.tf

### DISCUSSION THREAD

**Discussion by mauriciobls** [chgasparoto/curso-aws-com-terraform#152] [Q&A]
resource "aws_s3_bucket" "this" {
  bucket = var.name
  acl    = var.acl
  policy = var.policy
  tags   = var.tags

  dynamic "website" {
    for_each = length(keys(var.website)) == 0 ? [] : [var.website]
    content {
      index_document           = lookup(website.value, "index_document", null)
      error_document           = lookup(website.value, "error_document", null)
      redirect_all_requests_to = lookup(website.value, "redirect_all_requests_to", null)
      routing_rules            = lookup(website.value, "routing_rules", null)
    }
  }

  dynamic "versioning" {
    for_each = length(keys(var.versioning)) == 0 ? [] : [var.versioning]
    content {
      enabled    = lookup(versioning.value, "enabled", null)
      mfa_delete = lookup(versioning.value, "mfa_delete", null)
    }
  }
}

module "objects" {
  source = "./s3_object"

  for_each = var.files != "" ? fileset(var.files, "**") : []

  bucket = aws_s3_bucket.this.bucket
  key    = "${var.key_prefix}/${each.value}"
  src    = "${var.files}/${each.value}"
}

e o seguinte erro
faz todo o processo e não passa deste ponto
desde de já agradeço

with module.website.aws_s3_bucket.this,
│   on s3_module\main.tf line 1, in resource "aws_s3_bucket" "this":
│    1: resource "aws_s3_bucket" "this" {

**Comment by chgasparoto** [comment]
@mauriciobls consegue postar o erro inteiro? Não dá pra saber o que está dando de errado olhando para este pedaço de código.
